core/capabilities/launcher.go

```diff
@@ -92,66 +87,24 @@ func (w *launcher) Name() string {
 	return "CapabilitiesLauncher"
 }
 
-func (w *launcher) LocalNode(ctx context.Context) (capabilities.Node, error) {
-	if w.peerWrapper.GetPeer() == nil {
-		return w.localNode, errors.New("unable to get local node: peerWrapper hasn't started yet")
-	}
-
-	if w.localNode.WorkflowDON.ID == 0 {
-		return w.localNode, errors.New("unable to get local node: waiting for initial call from syncer")
-	}
-
-	return w.localNode, nil
-}
-
-func (w *launcher) updateLocalNode(state registrysyncer.State) {
-	pid := w.peerWrapper.GetPeer().ID()
-
-	var workflowDON capabilities.DON
-	capabilityDONs := []capabilities.DON{}
-	for _, d := range state.IDsToDONs {
-		for _, p := range d.NodeP2PIds {
-			if p == pid {
-				if d.AcceptsWorkflows {
-					if workflowDON.ID == 0 {
-						workflowDON = *toDONInfo(d)
-						w.lggr.Debug("Workflow DON identified: %+v", workflowDON)
-					} else {
-						w.lggr.Errorf("Configuration error: node %s belongs to more than one workflowDON", pid)
-					}
-				}
-
-				capabilityDONs = append(capabilityDONs, *toDONInfo(d))
-			}
-		}
-	}
-
-	w.localNode = capabilities.Node{
-		PeerID:         &pid,
-		WorkflowDON:    workflowDON,
-		CapabilityDONs: capabilityDONs,
-	}
-}
-
-func (w *launcher) Launch(ctx context.Context, state registrysyncer.State) error {
-	w.lggr.Debugw("running capabilities launcher", "state", state)
-	w.updateLocalNode(state)
+func (w *launcher) Launch(ctx context.Context, state *registrysyncer.LocalRegistry) error {
```

Comment:
Follow-up: `Launch` is called every time `Sync` cycle happens, so it is a bit of a misnomer. Find a better name (Handler?)